clang/lib/CodeGen/CGOpenMPRuntimeGPU.cpp

```diff
@@ -1099,13 +1197,23 @@ void CGOpenMPRuntimeGPU::emitNonSPMDKernel(const OMPExecutableDirective &D,
   // Reserve place for the globalized memory.
   GlobalizedRecords.emplace_back();
   if (!KernelStaticGlobalized) {
-    KernelStaticGlobalized = new llvm::GlobalVariable(
-        CGM.getModule(), CGM.VoidPtrTy, /*isConstant=*/false,
-        llvm::GlobalValue::InternalLinkage,
-        llvm::ConstantPointerNull::get(CGM.VoidPtrTy),
-        "_openmp_kernel_static_glob_rd$ptr", /*InsertBefore=*/nullptr,
-        llvm::GlobalValue::NotThreadLocal,
-        CGM.getContext().getTargetAddressSpace(LangAS::cuda_shared));
+    KernelStaticGlobalized =
+        (CGM.getTriple().isAMDGCN())
+            ? new llvm::GlobalVariable(
+                  CGM.getModule(), CGM.VoidPtrTy, /*isConstant=*/false,
+                  llvm::GlobalValue::WeakAnyLinkage,
```

Comment:
Why is InternalLinkage not working?

Reply:
Unclear. Works fine on gfx906, failed on gfx803 in difficult to debug fashion.

Reply:
Thinking of removing this unless you think otherwise. On gfx908, program compiles and run successfully with older version